### PR TITLE
Add service catalog REST API docs

### PR DESCRIPTION
**Pull request by vithu30**

### CODE REVIEW

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 1]
```diff
@@ -0,0 +1,1089 @@
...
+        content:
+          multipart/form-data:
+            schema:
+              required:
+                - file
+              properties:
+                file:
+                  type: string
+                  description: |
+                    Zip archive consisting of exported Application Configuration.
```
Comment (malinthaprasan) [inline en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml:465]: Seems like the description is incorrect. 
Shall we also add more details here. We can include the structure of the service folder, when there are multiple services present etc. Also mention the structure of a single folder (service). Give a short explanation about each file inside the folder too.
(can use `tree` command)
eg: 
```
Folder structure of the zip file should follow the below format:
<zip_file>.  (be specific whether we need to have a root folder here or just need to add all service folders inside the root level of the zip file)
├── service1
│   ├── metadata.yaml
│   └── definition.yaml
└── service2
    ├── metadata.yaml
    └── definition.yaml

**Here:**
**metadata.yaml** - service metadata of the service. This should adhere to the Service Schema.  
<we can put a example too>
**definition.yaml** - definition of the given service.
```

We can also mention about the Atomic behaviour of this operation too. i.e. if multiple services are present, if one service import fails, neither will get imported. Every service import needs to success in order to get the operation successful.

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 2]
```diff
@@ -0,0 +1,1089 @@
...
+                  format: binary
+                verifier:
+                  type: string
```
Comment (malinthaprasan) [inline en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml:468]: Shall we put a similar explanation to the verifier field too?

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 3]
```diff
@@ -0,0 +1,1089 @@
...
+          description: |
+            Successful response as the exported service as a zipped archive.
+          content:
+            application/zip:
+              schema:
+                type: string
+                format: binary
+          headers:
+            ETag:
+              $ref: '#/components/headers/ETag'
```
Comment (malinthaprasan) [inline en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml:535]: Let's remove the unsupported header.

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 4]
```diff
@@ -0,0 +1,1089 @@
...
+
+  ######################################################
+  # The Service Catalog API settings
+  ######################################################
+  /settings:
+    #-----------------------------------------------------
+    # Retrieve Service Catalog settings
+    #-----------------------------------------------------
+    get:
+      summary: Retrieve service catalog API settings
```
Comment (malinthaprasan) [inline en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml:555]:
```suggestion
      summary: Retrieve Service Catalog API settings
```
Let's keep all the `Service Catalog` entries consistent with a single form.

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 5]
```diff
@@ -0,0 +1,1089 @@
...
+    # The settings resource
+    #-----------------------------------------------------
+    Settings:
+      title: SettingsDTO
+      properties:
+        scopes:
+          type: array
+          items:
+            type: string
+            example: service_catalog:service_view
```
Comment (malinthaprasan) [inline en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml:833]: shall we add all the supported scopes into the example?

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 6]
```diff
@@ -0,0 +1,1089 @@
...
+        200:
+          description: |
+            Successful response with the imported service metadata.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ServiceInfoList'
+          headers:
+            ETag:
+              $ref: '#/components/headers/ETag'
```
Comment (malinthaprasan) [inline en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml:479]: I think we don't support this? Lets remove in that case.

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 7]
```diff
@@ -0,0 +1,1089 @@
...
+    #-------------------------------------------------------
+    # ifMatch - If-Match header value to be used for concurrency control
+    #-------------------------------------------------------
+    ifMatch:
+      name: If-Match
+      in: header
+      description: ETag of the service resource to update
+      required: false
+      schema:
+        type: string
```
Comment (malinthaprasan) [inline en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml:1031]: Remove the unused header.

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 8]
```diff
@@ -0,0 +1,1089 @@
...
+        - definitionFile
+      properties:
+        serviceMetadata:
+          $ref: '#/components/schemas/Service'
+        definitionFile:
+          type: string
+          format: binary
+        inlineContent:
+          type: string
+          description: Inline content of the document
```
[the remark this answers is not in the document]
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml]: Fixed in 0e2d05ca1eb804b7d1352c77d474e1c01accdbdd

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 9]
```diff
@@ -0,0 +1,1089 @@
...
+    Unauthorized:
+      description: Unauthorized. The user is not authorized.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+          example:
+            code: 401
+            message: Unauthorized
+            description: The user is not authorized
```
[the remark this answers is not in the document]
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml]: Fixed in 0e2d05ca1eb804b7d1352c77d474e1c01accdbdd

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 10]
```diff
@@ -0,0 +1,1089 @@
...
+        - lang: Curl
+          source: 'curl -k -H "Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8"
+            "https://127.0.0.1:9443/api/am/service-catalog/v0/services/3784c2da-2f2a-4af9-8a43-ef207fb4d739/definition"'
+
+  #-------------------------------------------------------------------
+  # Retrieve the information of APIs that use a given service
+  #-------------------------------------------------------------------
+  /services/{serviceId}/usage:
+    get:
+      summary: Retrieve the API Info that use the given service
```
[the remark this answers is not in the document]
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml]: Fixed in 0e2d05ca1eb804b7d1352c77d474e1c01accdbdd

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml (hunk 1 above)
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml]: Fixed in 0e2d05ca1eb804b7d1352c77d474e1c01accdbdd

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml (hunk 2 above)
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml:479]: Fixed in 0e2d05ca1eb804b7d1352c77d474e1c01accdbdd

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml (hunk 3 above)
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml]: Removed in 0e2d05ca1eb804b7d1352c77d474e1c01accdbdd

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml (hunk 6 above)
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml]: Yes, removed in 0e2d05ca1eb804b7d1352c77d474e1c01accdbdd

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml (hunk 7 above)
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml]: Fixed in 00f53c7

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml (hunk 5 above)
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml]: Fixed in 00f53c7

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml (hunk 4 above)
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml]: Fixed in 00f53c7

File: en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml [hunk 11]
```diff
@@ -0,0 +1,1089 @@
...
+          description: |
+            Successful response with the definition file as entity in the body.
+          content:
+            application/json:
+              schema:
+                type: string
+            application/yaml:
+              schema:
+                type: string
+                format: binary
```
[the remark this answers is not in the document]
Comment (vithu30) [reply en/docs/reference/product-apis/service-catalog-apis/service-catalog-v1/service-catalog-v1.yaml]: Fixed in 0e2d05ca1eb804b7d1352c77d474e1c01accdbdd. Removed `application/json` and added example using `externalValue`.

**Review by malinthaprasan — APPROVED** [review]

**Comment by vithu30** [comment]
> small typo

Fixed in 06316b1

**Pull request merged**